activemq-client/src/main/java/org/apache/activemq/ActiveMQMessageConsumer.java

```diff
@@ -589,6 +590,35 @@ public Message receive() throws JMSException {
         return createActiveMQMessage(md);
     }
 
+    /**
+     * JMS 2.0 support method for receiveBody(bodyType)
+     * AMQ-8464
+     */
+    <T> T receiveBody(Class<T> bodyType) throws JMSException {
+        checkClosed();
+        checkMessageListener();
+
+        sendPullCommand(0);
+        MessageDispatch md = dequeue(-1);
+        if (md == null) {
+            return null;
+        }
+
+        beforeMessageIsConsumed(md);
```

Comment:
This line is also seems like a problem as the spec says that the method has to act like the receive never happened in certain modes (auto akck and dups ok) and calling this has side effects such as adding to deliveredMessages and updating next delivery id